includes/attendee/attendee-repository.php

```diff
@@ -86,47 +86,77 @@ public function remove_attendee( int $event_id, int $user_id ): void {
 	/**
 	 * @throws Exception
 	 */
-	public function get_attendee( int $event_id, int $user_id ): ?Attendee {
+	public function get_attendee_for_event_for_user( int $event_id, int $user_id ): ?Attendee {
+		$attendees = $this->get_attendees_for_events_for_user( array( $event_id ), $user_id );
+		if ( empty( $attendees ) ) {
+			return null;
+		}
+		return $attendees[0];
+	}
+
+	/**
+	 * @var int[] $event_ids
+	 * @return Attendee[]
+	 * @throws Exception
+	 */
+	public function get_attendees_for_events_for_user( array $event_ids, int $user_id ): array {
+		if ( empty( $event_ids ) ) {
+			return array();
+		}
+
+		// Prevent SQL injection.
+		foreach ( $event_ids as $event_id ) {
+			if ( is_numeric( $event_id ) ) {
+				$event_id = intval( $event_id );
+			}
+			if ( ! is_int( $event_id ) || $event_id <= 0 ) {
+				return array();
+			}
+		}
+
 		global $wpdb, $gp_table_prefix;
+		$event_id_params = implode( ',', array_fill( 0, count( $event_ids ), '%d' ) );
 
 		// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
 		// phpcs:disable WordPress.DB.DirectDatabaseQuery.DirectQuery
 		// phpcs:disable WordPress.DB.DirectDatabaseQuery.NoCaching
-		$row = $wpdb->get_row(
+		$rows = $wpdb->get_results(
 			$wpdb->prepare(
 				"
-				select
-					user_id,
-					is_host,
-					is_new_contributor,
-					(
-						select group_concat( distinct locale )
-						from {$gp_table_prefix}event_actions
-						where event_id = attendees.event_id
-						  and user_id = attendees.user_id
-					) as locales
-				from {$gp_table_prefix}event_attendees attendees
-				where event_id = %d
-				  and user_id = %d
-			",
-				array(
-					$event_id,
-					$user_id,
+					select
+					    event_id,
+						user_id,
+						is_host,
+						is_new_contributor,
+						(
+							select group_concat( distinct locale )
+							from {$gp_table_prefix}event_actions
+							where event_id = attendees.event_id
+							  and user_id = attendees.user_id
+						) as locales
+					from {$gp_table_prefix}event_attendees attendees
+					where event_id in ($event_id_params)
+					  and user_id = %d
+				",
+				array_merge(
+					$event_ids,
+					array( $user_id ),
 				),
-			)
+			),
```

Comment:
Would we get a benefit from this? (We'd need to change the `$attendees[0]` above.

```suggestion
			OBJECT_K
			)
```

Otherwise we should be careful with the trailing comma in the function, this is pretty new syntax for php ([since 7.3](https://wiki.php.net/rfc/trailing-comma-function-calls)).

Reply:
Makes sense, that will allow us to not have to do [this](https://github.com/WordPress/wporg-gp-translation-events/pull/282/files#diff-706179ae733561a4c131b9d753beb83d4cc944de6de99f5c8beb3c0fdc4c1f9dR52-R55).